libc/stdlib/ftoa.c

```diff
@@ -0,0 +1,41 @@
+#include <math.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+int get_decimal_places(float number);
+
+char *ftoa(float number, int ndigit, char *buf) {
+  ndigit = fmin((uint32_t)ndigit, get_decimal_places(number));
```

Comment:
why did you convert ndigit to (uint32_t) but didn't convert get_decimal_places return value to (uint32_t) too?

Reply:
originally this was in order to discard negative ndigit values, but instead we it now takes it as a uint8_t type argument so it should be clearer.